### PR TITLE
[device/accton]: Update Accton-AS7712-32X

### PR DESCRIPTION
**Pull request by pollyhsu2git**
* Add Broadcom LED microprocessor / pre-emphasis initialization

Signed-off-by: polly_hsu@accton.com

### CODE REVIEW

File: platform/broadcom/docker-syncd-brcm/start.sh [hunk 1]
```diff
@@ -8,16 +8,22 @@ supervisorctl start rsyslogd
...
+while true; do
+    if [ -e /var/run/sswsyncd/sswsyncd.socket ]; then
+        break
+    fi
+    sleep 1
+done
+
 # If this platform has an initialization file for the Broadcom LED microprocessor, load it
 if [ -r ${PLATFORM_DIR}/led_proc_init.soc ]; then
     # Wait until syncd has created the socket for bcmcmd to connect to
```
Comment (qiluo-msft) [inline platform/broadcom/docker-syncd-brcm/start.sh]: ># Wait until syncd has cre [](start = 4, length = 26)

The comment is for the while loop you moved.
Suggest not to change the position of while loop. You may create a function, and call it before each bcmcmd.

File: device/accton/x86_64-accton_as7712_32x-r0/preemphasis_init.soc [hunk 2]
```diff
@@ -0,0 +1,802 @@
+echo "loading preemphasis..."
```
Comment (lguohan) [inline device/accton/x86_64-accton_as7712_32x-r0/preemphasis_init.soc]: can you specify these using config.bcm?

File: platform/broadcom/docker-syncd-brcm/start.sh (hunk 1 above)
Comment (pollyhsu2git) [inline platform/broadcom/docker-syncd-brcm/start.sh]: @qiluo-msft, I don't quite understand what you meant here~  Could you do me a favor to be specific it more?

File: device/accton/x86_64-accton_as7712_32x-r0/preemphasis_init.soc (hunk 2 above)
Comment (pollyhsu2git) [inline device/accton/x86_64-accton_as7712_32x-r0/preemphasis_init.soc]: @lguohan, unfortunately, config.bcm DOES NOT support the setting of TXFIR_POST2 / TXFIR_POST3.  That's why we put it within SOC.

File: platform/broadcom/docker-syncd-brcm/start.sh (hunk 1 above)
Comment (qiluo-msft) [inline platform/broadcom/docker-syncd-brcm/start.sh]: (updated original comment, please check)

Comment (pollyhsu2git) [inline platform/broadcom/docker-syncd-brcm/start.sh]: @qiluo-msft, I got your point about the comment (1), but for (2), still need your help to clarify it further.

Comment (qiluo-msft) [inline platform/broadcom/docker-syncd-brcm/start.sh]: (updated original comment again, please check)

**Comment by pollyhsu2git** [comment]
@lguohan, the preemphsis code was removed done~

**Comment by pollyhsu2git** [comment]
@lguohan, do you have any further comment?  We are waiting this merge for testing.

**Comment by pollyhsu2git** [comment]
@lguohan,  do you have any further concerns? We are waiting this merge into the git master for testing.

**Comment by pollyhsu2git** [comment]
@lguohan / @stcheng,  this PR has been in the queue for one month~ Do you have any further concerns? What can we do to speed up the PR process?

**Comment by pollyhsu2git** [comment]
@stcheng, as a reminder, we've sent you the updated *config.bcm* (th-as7712-32x100G.config.bcm) which we added the pre-emphasis support for AS7712-32X via your Microsoft e-mail address with the subject,*FW: [Azure/sonic-buildimage] [device/accton]: Update Accton-AS7712-32X (#776) ---> th-as7312-48x25G+6x100G.config.bcm* on 2017-09-26.

**Comment by pollyhsu2git** [comment]
@stcheng, we've checked your latest libsaibcm_3.0.3.2-5_amd64.deb (jenkins#360 built on 2017-10-15), and th-as7712-32x100G.config.bcm is not yet updated.

**Pull request merged**